src/controllers/update/beneficial.owner.type.controller.ts

```diff
@@ -41,9 +43,64 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
+const fetchAndSetBoMo = async (req: Request, appData: ApplicationData) => {
+  if (!hasFetchedBoAndMoData(appData)) {
+    if (!appData.update) {
+      appData.update = {};
+    }
+    appData.update.review_beneficial_owners_individual = [];
+    appData.update.review_beneficial_owners_corporate = [];
+    appData.update.review_beneficial_owners_government_or_public_authority = [];
+
+    await retrieveBeneficialOwners(req, appData);
+    await retrieveManagingOfficers(req, appData);
+    setFetchedBoMoData(appData);
+  }
+};
+
+const checkBOIndividualValidation = (boi: BeneficialOwnerIndividual) => {
+  if (boi.usual_residential_address){
+    return true;
+  } else {
+    return false;
+  }
+};
+
+export const checkAndReviewBeneficialOwner = (appData: ApplicationData) => {
+  let redirectUrl = "";
+  const beneficialOwnerReviewRedirectUrl = `${config.UPDATE_AN_OVERSEAS_ENTITY_URL
+    + config.UPDATE_REVIEW_BENEFICIAL_OWNER_INDIVIDUAL_PAGE
+    + config.REVIEW_BENEFICIAL_OWNER_INDEX_PARAM}`;
+
+  // Check last individual BO validates - in case back button is clicked
+  const boiLength = appData.beneficial_owners_individual?.length || 0;
+  const boiIndex: number = boiLength - 1;
+  if ((appData.beneficial_owners_individual && boiLength >= 1) && !checkBOIndividualValidation(appData.beneficial_owners_individual[boiIndex])) {
+    redirectUrl = `${beneficialOwnerReviewRedirectUrl}${boiIndex}`;
+    return redirectUrl;
+  }
+
+  if (boiLength >= 0){
+    const boi = appData.update?.review_beneficial_owners_individual?.pop() as BeneficialOwnerIndividual;
+    if (!boi){
```

Comment:
should this redirect to the  BO/MO summary page (and not the BOI review)?

Reply:
According to the AC below- it should go to summary page if no more "bo" to review

"Successfully continuing from this page will bring the user to the next available beneficial owner, or if none are available, the summary page***"

Reply:
I don't think line 83 should be >0, the first time, it comes for review, appData.beneficialOwner.length will be 0. If it is to be >0, then the whole block won't execute in the first place

Reply:
As I said, I don't think it will be !boi after pop but better to do something sensible like throw an error
if that's wrong